### ISSUE TITLE
Improve an error message in qiskit.converters.circuit_to_gate()

### ISSUE THREAD

**Issue by hay-k** [Qiskit/qiskit-terra#4064]
### What is the expected enhancement?

Let's assume we have `QuantumCircuit` object called `qc`, and one tries to convert it into a `Gate` object using `qiskit.converters.circuit_to_gate()`. If `qc` contains some instructions which cannot be converted into `Gate`, the following exception is raised
```
QiskitError: 'One or more instructions in this instruction cannot be converted to a gate'
```
My suggestion is to improve this error message and add some info about the particular instruction preventing the convertion from happening. I believe, something like the instruction name in the error message should be more helpfull, than the current general statement.

Below is a code snippet (for a `qc` containing a measurement operation) which can be used to achieve the error mentioned above
```
from qiskit import QuantumCircuit
from qiskit.converters import circuit_to_gate

qc = QuantumCircuit(1, 1)
qc.h(0)
qc.measure(0, 0)

gate = circuit_to_gate(qc)
```

**Comment by enavarro51** [comment]
@hay-k and @kdk, I'm new to qiskit and would like to give this a try.

**Comment by enavarro51** [comment]
@hay-k. I got a different error from the code snippet above. 'Circuit with classical bits cannot be converted to gate.' This is trapped above the other one and is because there's a cbit. I used qc.barrier() and that caught the error you had. My plan is to just add the name of the first instruction that fails and let the user iteratively fix things if there are multiple such instructions.
Something like 'One or more instructions cannot be converted to a gate. "barrier" is not a gate instruction'
Does this sound like what you're looking for? Thanks.
